### PR TITLE
Refactor exo test

### PR DESCRIPTION
**Pull request by alexdavid**
First step in my general refactor to help clean up the tight coupling in exosphere. To keep PRs small this initial refactor moves most of the mess into a single file (service_tester), which will be removed in my next PR.

It also introduces the idea of an `AppContext` and a `ServiceContext` which is a way of telling helper methods and docker compose about an app or a service.

### CODE REVIEW

File: src/application/service_tester.go [hunk 1]
```diff
@@ -16,12 +17,42 @@ type ServiceTester struct {
...
 
 // NewServiceTester is ServiceTester's constructor
-func NewServiceTester(role string, serviceConfig types.ServiceConfig, builtDependencies map[string]config.AppDevelopmentDependency, appDir, serviceDir string, initializer *Initializer, runner *Runner) (*ServiceTester, error) {
+func NewServiceTester(serviceContext types.ServiceContext, logger *util.Logger, mode composebuilder.BuildMode) (*ServiceTester, error) {
+	appConfig := serviceContext.AppContext.Config
+	serviceConfig := serviceContext.Config
+	serviceDir := serviceContext.Location
+	role := serviceContext.Name
+	appDir := serviceContext.AppContext.Location
+	dockerComposeProjectName := fmt.Sprintf("%stests", composebuilder.GetDockerComposeProjectName(appDir))
```
Comment (hugoesthere) [inline src/application/service_tester.go:31]: Can we still get this from a shared function like `getTestDockerComposeProjectName`? I think `exo clean` may need it.

File: src/types/context.go [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+// Context represents contextual information about what application/service the user is running
+type Context struct {
+	AppContext        AppContext
+	ServiceContext    ServiceContext
+	HasServiceContext bool
+}
+
+// GetContext returns a Context for the current working direcotry
+func GetContext() (Context, error) {
```
Comment (hugoesthere) [inline src/types/context.go:13]: 🏅

File: src/types/service_context.go [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+	yaml "gopkg.in/yaml.v2"
+)
+
+// ServiceContext represents the exosphere service the user is running
+type ServiceContext struct {
+	Name       string
+	Location   string
+	Config     ServiceConfig
+	AppContext AppContext
+
```
Comment (hugoesthere) [inline src/types/service_context.go]: extra line

**Review by hugoesthere — APPROVED** [review]
Looks good to me. I like populating all the context fields in `application/...` more than when we used to do it in `cmd/...`. Hope this plays well with the rest of the commands

File: src/application/service_tester.go (hunk 1 above)
Comment (alexdavid) [reply src/application/service_tester.go:31]: Yeah I haven't cleaned up the service tester yet. Ideally I don't want anything in application to know about docker compose stuff. The docker package should actually take care of this IMO

**Pull request merged**